utils/demangle.c

```diff
@@ -1426,19 +1427,30 @@ static int dd_source_name(struct demangle_data *dd)
 	/* check special symbol mappings (e.g. '$LT$') for Rust */
 	while (dollar != NULL && dollar < end) {
 		bool found = false;
-
 		num = dollar - p;
-		dd_append_len(dd, p, num);
+		/* consume extra '_' before '$' */
```

Comment:
I'm not sure about this.  How can we make sure if it's not a valid identifier.  For example it might have `my_mod_` as a module name, then any dollar sign after that would delete the trailing underscore, right?

Reply:
That's a good point. I'll revert this code. However, for clearer context, the first underscore would be deleted IMO.